pkg/kubelet/kubelet_pods.go

```diff
@@ -538,6 +538,11 @@ func (kl *Kubelet) getServiceEnvVarMap(ns string) (map[string]string, error) {
 		//
 		// ordering of the case clauses below enforces this
 		case ns:
+			// If service links are not enabled, we shouldn't add services from
+			// the same namespace.
+			if !*enableServiceLinks {
```

Comment:
I think we need to special-case KUBERNETES_* vars (master) because clients need it.